tls/s2n_security_policies.h

```diff
@@ -25,12 +25,62 @@
 /* Kept up-to-date by s2n_security_policies_test */
 #define NUM_RSA_PSS_SCHEMES 6
 
+/* The s2n_security_policy struct is used to define acceptable and available
+ * algorithms for use in the TLS protocol. Note that the behavior of each field
+ * likely differs between different TLS versions, as the mechanics of cipher
+ * negotiation often have significant differences between TLS versions.
+ *
+ *= https://www.rfc-editor.org/rfc/rfc5246#section-7.4.2
+ *= type=exception
+ *= reason=not implemented due to lack of utility
+ *# If the client provided a "signature_algorithms" extension, then all
+ *# certificates provided by the server MUST be signed by a
+ *# hash/signature algorithm pair that appears in that extension.
+ *
+ *= https://www.rfc-editor.org/rfc/rfc8446#section-4.2.3
+ *= type=exception
+ *= reason=not implemented due to lack of utility
+ *# If no "signature_algorithms_cert" extension is present, then the
+ *# "signature_algorithms" extension also applies to signatures appearing in
+ *# certificates.
+ *
+ * In s2n-tls, the signature_algorithms extension only applies to signatures in
+ * CertificateVerify messages. To specify acceptable signature algorithms for
+ * certificates the certificate_signature_preferences field should be set in the
+ * security policy.
+ */
 struct s2n_security_policy {
     uint8_t minimum_protocol_version;
+    /* TLS 1.0 - 1.2 - cipher preference includes multiple elements such
+     * as signature algorithms, record algorithms, and key exchange algorithms
+     * TLS 1.3 - cipher preference only determines record encryption
+     */
     const struct s2n_cipher_preferences *cipher_preferences;
+    /* kem_preferences is only used for Post-Quantum cryptography */
     const struct s2n_kem_preferences *kem_preferences;
+    /* This field roughly corresponds to the "signature_algorithms" extension.
+     * The client serializes this field of the security_policy to populate the
+     * extension, and it is also used by the server to choose an appropriate
+     * entry from the options supplied by the client.
+     * TLS 1.2 - optional extension to specify signature algorithms other than
+     * default: https://www.rfc-editor.org/rfc/rfc5246#section-7.4.1.4.1
+     * TLS 1.3 - required extension specifying signature algorithms
+    */
     const struct s2n_signature_preferences *signature_preferences;
+    /* When this field is set, the endpoint will ensure that the peer's
+     * certificate signature algorithm is in the specified list. Note that this
+     * list does not correspond to the signature_algorithms_cert extension.
+     * There is no scenario in which this information is transmitted to a peer.
```

Comment:
If you're going to write this comment, you need to be VERY correct about what the list applies to, because that is the biggest source of confusion. It's the signatures on the certificates in the cert chain, not any signature calculated during the handshake.
```suggestion
    /* When this field is set, the endpoint will ensure that the signatures on the certificates
     * in the peer's certificate chain are in the specified list. Note that this
     * list does not correspond to the signature_algorithms_cert extension.
     * There is no scenario in which this information is transmitted to a peer.
```

Reply:
Whoops, that was what I meant to communicate. Mentally I've just taken to talking about them as "transcript signatures" and "certificate signature" but it's a good callout that the comment is a bit confusing. And definitely a miss in not emphasizing the "certificate chain" aspect.